src/librustc_yk_sections/mir_cfg.rs

```diff
@@ -127,6 +128,10 @@ pub fn generate_tir<'a, 'tcx, 'gcx>(
     let mut sorted_def_ids: Vec<&DefId> = def_ids.iter().collect();
     sorted_def_ids.sort();
 
+    // Only if the user requested, open a file for the textual TIR dump.
+    let mut dump_file: Option<File> = dump_filename.map_or_else::<Result<Option<File>, io::Error>, _, _>(
```

Comment:
Are the type annotations needed? They make this very hard to read.

Reply:
Hah, ironically I added them because I found it hard to read.

I'm not proud of this line. I think I should unroll it into something simpler.

Reply:
I'd just drop the annotations myself and then I think it'll be fine.